spec/namespaces.md

```diff
@@ -36,12 +36,16 @@ The two compilation units contribute to the single global namespace, in this cas
 
 ## Namespace declarations
 
-A *namespace_declaration* consists of the keyword `namespace`, followed by a namespace name and body, optionally followed by a semicolon.
+A *namespace_declaration* has two different forms.  The first consists of the keyword `namespace`, followed by a namespace name and body, optionally followed by a semicolon.  The second consists of the keyword `namespace`, followed by a namespace name, a semicolon and an optional list of directives and declarations.
 
 ```antlr
 namespace_declaration
     : 'namespace' qualified_identifier namespace_body ';'?
     ;
+    
+single_line_namespace_declaration
+    : 'namespace' qualified_identifier ';' extern_alias_directive* using_directive* type_declaration*
```

Comment:
Consider how top-level statements fit in here as well, since they're in a speclet currently, not in the main spec.

Reply:
can you link me to that speclet?